Automattic/jetpack
# Add geo-location support

Fixes #7321

#### Changes proposed in this Pull Request:

* This PR adds support for the "geo-location" feature on both pages and posts.  Calypso offers a UI for setting the necessary meta fields but it currently only works with WP.com sites.  

* This also leverages several hooks to make use of the geo-location data when possible: during RSS feed generation, in meta tags in `wp_head`, and by outputting Geo microformat data and a visible location label on single post pages, just after the normal content.  These features bring Jetpack in line with the equivalent features from WP.com. 

#### Testing instructions:

Because Calypso currently excludes Jetpack sites from using the geo-location feature, you'll need to test by running this branch (https://github.com/Automattic/wp-calypso/tree/add/let-jetpack-use-geolocation).  You can also manually set the necessary post meta values via the REST API, if that's easier.  The plugin is looking for the presence of the following meta fields: geo_public, geo_latitude, geo_longitude, and geo_address.

Once this PR is ready to be merged into a Jetpack release, we can modify Calypso to include a minimum version check, allowing Jetpack users to use this feature.

Once you have geo-location data added on your post, you can test that it is included in RSS, in `wp_head` and on post content (when viewing a single post).  

There are also fairly extensive unit tests included in this PR that cover concerns like hiding the geo-location data when it is marked as private and output escaping.

#### Proposed changelog entry for your changes:

* Support display of geo-location data added to posts and pages with Calypso.

Thanks in advance for taking the time to look this over!  Let me know if there is anything I can do to clarify or assist with testing.  I look forward to your feedback!
